jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/DeploymentScanner.java

```diff
@@ -1185,36 +1188,72 @@ public void setContainerScanJarPattern(String pattern)
         }
 
         /**
-         * The name of the class that this environment uses to create {@link ContextHandler}
-         * instances (supports a class that implements {@code java.util.function.Supplier<Handler>} as well).
+         * The class that this environment uses to create {@link Deployable.ClassLoaderFactory} instances.
+         *
+         * <p>
+         * This is the class used to create a ClassLoader for the environment before
+         * any XML files are loaded to create/configure the {@link ContextHandler}.
+         * </p>
+         *
+         * @param classLoaderFactoryClass the class for this environment's class loader factory.
+         * @see StandardContextHandlerFactory#CLASSLOADER_FACTORY_ATTRIBUTE
+         * @see org.eclipse.jetty.server.Deployable.ClassLoaderFactory
+         */
+        public void setClassLoaderFactoryClass(Class<? extends Deployable.ClassLoaderFactory> classLoaderFactoryClass)
```

Comment:
I don't like this being exposed so that users could configure anything here, even for webapps. Webapps have a very structured way of creating their classloaders via the `Configuration` classes and any override of that here would cause havoc. 

It seems the problem that is trying to be solved here is related solely to core deployments, so the solution should affect only  core deployment related classes.

Reply:
The ClassLoaderFactory has been removed.